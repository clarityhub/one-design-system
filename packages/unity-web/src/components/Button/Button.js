import React from 'react';
import cx from 'classnames';
import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { bool, func, oneOf, oneOfType, node, string } from 'prop-types';
import { variants } from '@clarityhub/unity-core/lib/typography';
import borders from '@clarityhub/unity-core/lib/borders';
import colors from '@clarityhub/unity-core/lib/colors';
import { lighten, darken, desaturate } from '@clarityhub/unity-core/lib/utilities/color';

import Loading from '../svgs/Loading';

const Button = styled.button(
	css`
        overflow: visible;
        display: inline-block;

        ${variants.button.string}

        box-shadow: ${colors.shadow.default};
        cursor: pointer;
        text-align: center;
        white-space: nowrap;
        vertical-align: middle;
        user-select: none;
        border: 1px solid transparent;
        padding: .5rem 1.25rem;
        font-size: 1rem;
        line-height: 1.5;
        border-radius: ${borders.borderRadius.rem}rem;
        position: relative;
        text-decoration: none;
        text-transform: uppercase;
        transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;

        &:focus,
        &:hover {
            text-decoration: none;
        }

        &:focus,
        &.focused {
            outline: 0;
            box-shadow: 0 0 0 .2rem ${colors.focused.default};
        }

        &:disabled,
        &.disabled {
            cursor: not-allowed;
            pointer-events: none;
        }

        &:not(:disabled):not(.disabled) {
            cursor: pointer;
        }

        svg {
            height: 1rem;
            width: 1rem;
        }

        --text: ${colors.primary.default};
        --text-active: ${colors.black.default};
        --bg: transparent;
        --bg-active: transparent;
        --bg-disabled: transparent;

        &.-default,
        &.-primary,
        &.-danger,
        &.-white {
            color: var(--text);
            background-color: var(--bg);
            border-color: var(--bg);

            &.-outline {
                border-color: var(--outline, var(--bg));
                color: var(--outline-text, var(--bg));
            }

            &.-text {
                color: var(--text-text, var(--bg));
            }

            &:hover,
            &.active {
                background-color: var(--bg-active);
                border-color: var(--bg-active);
                color: var(--text-active);

                &.-outline {
                    background-color: var(--bg) !important;
                    border: 1px solid var(--bg);
                    color: var(--outline-text-active, var(--text));
                }

                &.-text {
                    color: var(--text-text-active, var(--bg-active));
                }
            }

            &:disabled,
            &.disabled {
                background-color: var(--bg-disabled);
                border-color: var(--bg-disabled);
                color: var(--text);

                &.-outline {
                    border: 1px solid var(--outline-disabled, var(--outline-text-disabled, var(--bg-disabled)));
                    color: var(--outline-text-disabled, var(--bg-disabled));
                }

                &.-text {
                    color: var(--text-text-disabled, var(--bg-disabled));
                }
            }
        }

        &.-primary {
            --text: ${colors.white.default};
            --text-active: ${colors.white.default};
            --outline-disabled: ${colors.gray.default};
            --bg: ${colors.primary.default};
            --bg-active: ${desaturate(darken(colors.primary.default))};
            --bg-disabled: ${desaturate(lighten(colors.primary.default))};
        }

        &.-danger {
            --text: ${colors.white.default};
            --text-active: ${colors.white.default};
            --outline-disabled: ${colors.gray.default};
            --bg: ${colors.danger.default};
            --bg-active: ${desaturate(darken(colors.danger.default))};
            --bg-disabled: ${lighten(colors.danger.default, 0.3)};
        }

        &.-default {
            --text: ${colors.black.default};
            --text-active: ${colors.black.default};
            --outline: ${colors.black.default};
            --outline-text: ${colors.black.default};
            --outline-text-active: ${colors.black.default};
            --outline-text-disabled: ${colors.gray.default};
            --text-text: ${colors.black.default};
            --text-text-active: ${colors.black.default};
            --text-text-disabled: ${colors.gray.default};
            --bg: ${colors.white.default};
            --bg-active: ${colors.dove.default};
            --bg-disabled: ${colors.muted.default};
        }

        &.-white {
            --text: ${colors.primary.default};
            --text-active: ${colors.black.default};
            --outline: ${colors.white.default};
            --outline-text: ${colors.white.default};
            --outline-text-active: ${colors.primary.default};
            --outline-text-disabled: ${colors.gray.default};
            --text-text: ${colors.white.default};
            --text-text-active: ${colors.white.default};
            --text-text-disabled: ${colors.gray.default};
            --bg: ${colors.white.default};
            --bg-active: ${colors.dove.default};
            --bg-disabled: ${colors.muted.default};
        }

        &.-text {
            background-color: transparent !important;
            border: 1px solid transparent;
            box-shadow: none;

            &:hover,
            &.active {
                border-color: transparent;
            }

            &.disabled,
            &:disabled {
                border: 1px solid transparent;
            }
        }

        &.-outline {
            background-color: transparent !important;
            border: 1px solid inherit;
            box-shadow: none;
        }

        &.disabled,
        &:disabled {
            cursor: not-allowed;
        }
    `,
	({ size }) => {
		switch (size) {
		case 'small':
			return css`
                font-size: 0.9rem;
                padding: .2rem 1rem;
            `;
		case 'large':
			return css`
                font-size: 1.2rem;
                padding: 0.8rem 1.4rem;
            `;
		case 'default':
		default:
			return '';
		}
	},

	({ block }) => block && css`
        display: block;
        width: 100%;
    `,

	({ loading }) => loading && css`
        color: transparent !important;
        cursor: wait !important;

        &:hover {
            color: transparent !important;
        }
    `,
);

const IconWrapper = styled.span`
    font-size: inherit;
    margin-right: 0.3rem;
    vertical-align: middle;

    svg {
        height: 1rem;
        width: 1rem;

        ${({ size }) => {
		switch (size) {
		case 'small':
			return css`
                height: 0.9rem;
                width: 0.9rem;
            `;
		case 'large':
			return css`
                height: 1.2rem;
                width: 1.2rem;
            `;
		case 'default':
		default:
			return '';
		}
	}}
    }
`;

const LoadingWrapper = styled.div`
    position: absolute;
    left: 50%;
    transform: translate(-50%, -50%);
    top: 50%;
    align-content: center;
    justify-content: center;
    display: flex;

    svg {
        height: 1.4rem;
        width: 1.4rem;

        ${({ size }) => {
		switch (size) {
		case 'small':
			return css`
                height: 1.2rem;
                width: 1.2rem;
            `;
		case 'large':
			return css`
                height: 2.4rem;
                width: 2.4rem;
            `;
		case 'default':
		default:
			return '';
		}
	}}

    path {
        ${({ type, outline, text }) => {
		switch (type) {
		case 'primary':
			return css`
                fill: ${colors.white.default};

                ${outline && css`
                    fill: ${colors.primary.default};
                `}

                ${text && css`
                    fill: ${colors.primary.default};
                `}

                &:hover,
                &.active {
                    ${outline && css`
                        fill: ${colors.white.default};
                    `}

                    ${text && css`
                        fill: ${desaturate(darken(colors.primary.default))};
                    `}
                }

                &:disabled,
                &.disabled {
                    ${outline && css`
                        fill: ${desaturate(lighten(colors.primary.default))};
                    `}

                    ${text && css`
                        fill: ${desaturate(lighten(colors.primary.default))};
                    `}
                }
            `;
		case 'danger':
			return css`
                fill: ${colors.white.default};

                ${outline && css`
                    fill: ${colors.danger.default};
                `}

                ${text && css`
                    fill: ${colors.danger.default};
                `}

                &:hover,
                &.active {
                    ${outline && css`
                        fill: ${colors.white.default};
                    `}

                    ${text && css`
                        fill: ${desaturate(darken(colors.danger.default))};
                    `}
                }

                &:disabled,
                &.disabled {
                    ${outline && css`
                        fill: ${desaturate(lighten(colors.danger.default, 0.3))};
                    `}

                    ${text && css`
                        fill: ${desaturate(lighten(colors.danger.default, 0.3))};
                    `}
                }
            `;
		case 'default':
		default:
			return css`
                fill: ${colors.black.default};

                ${outline && css`
                    fill: ${colors.black.default};
                `}

                ${text && css`
                    fill: ${colors.black.default};
                `}

                &:hover,
                &.active {
                    ${outline && css`
                        fill: ${colors.black.default};
                    `}

                    ${text && css`
                        fill: ${desaturate(darken(colors.black.default))};
                    `}
                }

                &:disabled,
                &.disabled {
                    fill: ${darken(colors.gray.default, 0.1)};

                    ${outline && css`
                        fill: ${lighten(colors.gray.default, 0.01)};
                    `}

                    ${text && css`
                        fill: ${desaturate(lighten(colors.black.default))};
                    `}
                }
            `;
		}}
}
        }
    }
`;

const renderIcon = (Icon) => {
	if (typeof Icon === 'function') {
		return <Icon />;
	}

	return Icon;
};

const ButtonWrapper = ({
	className,
	type,
	outline,
	text,
	size,
	icon,
	children, 
	component,
	loading,
	buttonType = 'button',
	...rest
}) => {
	let Component = Button;

	if (component) {
		Component = Button.withComponent(component);
	}

	return (
		<Component
			className={cx(className, {
				[`-${type}`]: type,
				'-outline': outline,
				'-text': text,
			})}
			outline={outline}
			text={text}
			size={size}
			loading={loading}
			type={buttonType}
			{...rest}
		>
			{
				icon && (
					<IconWrapper size={size}>{renderIcon(icon)}</IconWrapper>
				)
			}
			{
				loading && (
					<LoadingWrapper
						type={type}
						outline={outline}
						text={text}
						size={size}
						{...rest}
					>
						<Loading />
					</LoadingWrapper>
				)
			}
			{children}
		</Component>
	);
};

ButtonWrapper.propTypes = {
	block: bool,
	buttonType: string,
	component: oneOfType([string, func]),
	icon: oneOfType([node, func]),
	loading: bool,
	outline: bool,
	size: oneOf(['small', 'default', 'large']),
	text: bool,
	type: oneOf(['default', 'white', 'primary', 'danger', 'success']),
};

ButtonWrapper.defaultProps = {
	type: 'default',
};

export { Button as EmotionButton };
export default ButtonWrapper;
