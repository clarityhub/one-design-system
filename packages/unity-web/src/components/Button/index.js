export { default, EmotionButton } from './Button';