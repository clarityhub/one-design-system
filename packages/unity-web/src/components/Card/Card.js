import styled from '@emotion/styled';
import isPropValid from '@emotion/is-prop-valid';
import { css } from '@emotion/core';
import borders from '@clarityhub/unity-core/lib/borders';
import colors from '@clarityhub/unity-core/lib/colors';
import { bool, oneOf } from 'prop-types';

import { CardHeaderWrapper } from './CardHeader';
import CardBody from './CardBody';

const Card = styled('div', {
	shouldForwardProp: prop => isPropValid(prop) && !['type'].includes(prop),
})(
	css`
        position: relative;
        display: flex;
        flex-direction: column;
        min-width: 0;
        word-wrap: break-word;
        background-clip: border-box;
        border-radius: ${borders.borderRadius.rem}rem;
    `,
	({ flat }) => !flat ? css`
        box-shadow: ${colors.shadow.default};
    ` : css`
        box-shadow: 0 0 0 1px rgba(0,0,0,0);
    `,
	({ type }) => {
		switch (type) {
		case 'notification':
			return css`
                background: ${colors.gradient.default};

                ${CardBody} {
                    padding-top: 2rem;
                    padding-bottom: 2rem;
                }
            `;
		case 'highlight':
			return css`
                background: ${colors.highlight.default};
            `;
		case 'callout':
			return css`
                background: #F9FAFC;
                border: 1px solid ${colors.primary.default};
            `;
		default:
			return css`
                background-color: ${colors.white.default};
            `;
		}
	},
	css`
        > :not(${CardHeaderWrapper}) {
            h3 {
                margin: 0;
                padding: 0;
            }

            h4 {
                margin-top: 0.6rem;
            }

            p:only-child {
                margin: 0;
            }
        }
    `,
	({ center }) => center && css`
        text-align: center;
    `,
	({ hoverable }) => hoverable && css`
        transition: box-shadow 0.2s ease-in-out;

        &:hover {
            box-shadow: 0 0 0 1px rgba(50,50,93,.1),
                0 7px 14px 0 rgba(50,50,93,.1),
                0 3px 6px 0 rgba(0,0,0,.07)
        }
    `,
);

/**
 * Hack for docz and react-docgen
 */
const CardPropTypes = () => <div />;
CardPropTypes.propTypes = {
	center: bool,
	flat: bool,
	hoverable: bool,
	type: oneOf(['notification', 'highlight', 'callout']),
};
CardPropTypes.defaultProps = {
	center: false,
	hoverable: false,
	flat: false,
};
Card.propTypes = CardPropTypes.propTypes;
export { CardPropTypes };

export default Card;
