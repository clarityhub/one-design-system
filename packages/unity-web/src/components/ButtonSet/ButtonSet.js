import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { bool } from 'prop-types';
import { EmotionButton } from '../Button';
import Space from '../../scaffolding/Space';

const ButtonSet = styled(Space)`
    ${({ spread }) => spread && css`
        display: flex;
        align-items: center;
        justify-content: flex-end;

        ${EmotionButton} {
            &:not(:first-of-type) {
                margin-left: auto!important;
            }
        }
    `}

`;

/**
 * Hack for docz and react-docgen
 */
const ButtonSetPropTypes = () => <div />;
ButtonSetPropTypes.propTypes = {
	spread: bool,
};
ButtonSet.propTypes = ButtonSetPropTypes.propTypes;
export { ButtonSetPropTypes };

export default ButtonSet;

