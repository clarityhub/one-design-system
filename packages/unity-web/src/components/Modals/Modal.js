import React, { Component } from 'react';
import { node, bool, func, string } from 'prop-types';
import styled from '@emotion/styled';
import { css } from '@emotion/core';
import { opacify } from '@clarityhub/unity-core/lib/utilities/color';
import colors from '@clarityhub/unity-core/lib/colors';

import zIndex from '../../theme/zIndexes';
import Card from '../Card';
import ModalPortal from './ModalPortal';

// XXX refactor animations to be reuseable

const ModalWrapper = styled.div`
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    background-color: ${opacify(colors.dark.default, 0.6)};
    display: flex;
    z-index: ${zIndex.modal};

    ${({ open }) => !open && css`
        display: none;
    `}
`;

const ModalDialog = styled.div`
    transition: all .3s ease-out;
    transition-property: opacity, transform;
    transform: translate(0, -25%);
    display: flex;
    min-height: calc(100% - (.5rem * 2));
    opacity: 0;
    align-items: center;
    max-width: 500px;
    margin: 1.75rem auto;
    position: relative;
    width: 100%;

    @media screen and (prefers-reduced-motion:reduce) {
        transition: none;
    }

    ${({ open }) => open && css`
        opacity: 1;
        transform: translate(0, 0);
    `}

    ${Card} {
        position: relative;
        display: flex;
        flex-direction: column;
        width: 100%;
        pointer-events: auto;
    }
`;

export default class Modal extends Component {
    static propTypes = {
    	/**
		 * Passthrough to Card
		 */
    	center: bool,
    	/**
		 * Passthrough to Card
		 */
    	children: node,
    	onClose: func,
    	open: bool.isRequired,
    	type: string,
    }

    state = {
    	delayedOpen: false,
    }

    componentDidMount() {
    	this.setup();
    }

    componentDidUpdate(prevProps) {
    	if (!prevProps.open && this.props.open) {
    		this.setup();
    	}
        
    	if (prevProps.open && !this.props.open) {
    		this.teardown();
    	}
    }

    componentWillUnmount() {
    	this.teardown(true);
    }

    setup = () => {
    	this._onClickOutside = this.onClickOutside.bind(this);
    	this._onKeyUp = this.onKeyUp.bind(this);

    	document.addEventListener('click', this._onClickOutside);
    	window.addEventListener('keyup', this._onKeyUp);

    	this.checkOpen();
    }

    teardown = (immediate = false) => {
    	if (this.openTimer) {
    		clearTimeout(this.openTimer);
    	}

    	document.removeEventListener('click', this._onClickOutside);
    	window.removeEventListener('keyup', this._onKeyUp);
        
    	this.setState({
    		delayedOpen: false,
    	});
        
    	if (!immediate) {
    		this.closeTimer = setTimeout(() => {
    			this.setState({
    				delayedClose: false,
    			});
    		}, 100);
    	}
    }

    checkOpen = () => {
    	if (this.props.open && !this.state.delayedOpen) {
    		this.doneOpening = false;

    		if (this.openTimer) {
    			clearTimeout(this.openTimer);
    		}
            
    		this.setState({
    			delayedClose: true,
    		});

    		this.openTimer = setTimeout(() => {
    			this.doneOpening = true;
    			this.setState({
    				delayedOpen: true,
    			});
    		}, 100);
    	}
    }

    onClickOutside = (e) => {
    	if (this.modal && this.doneOpening) {
    		if (this.props.open) {
    			const target = e.target;
    			// check if target is within (or is) the modal
    			const clickOutsideModal = this.modal !== target && !this.modal.contains(target);
    			if (clickOutsideModal) {
    				this.close(e, { reason: 'onClickOutside' });
    			}
    		}
    	}
    }

    onKeyUp = (e) => {
    	if (this.props.open) {
    		if (e.key === 'Escape') {
    			this.close(e, { reason: 'escape' });
    		}
    	}
    }

    close = (...args) => {
    	const { onClose } = this.props;

    	if (onClose) {
    		onClose(...args);
    	}
    }

    render() {
    	const { center, children, type } = this.props;
    	const { delayedOpen, delayedClose } = this.state;
    
    	return (
    		<ModalPortal>
    			<ModalWrapper open={delayedClose}>
    				<ModalDialog open={delayedOpen}>
    					<Card ref={ref => this.modal = ref} type={type} center={center}>
    						{children}
    					</Card>
    				</ModalDialog>
    			</ModalWrapper>
    		</ModalPortal>
    	);
    }
}
