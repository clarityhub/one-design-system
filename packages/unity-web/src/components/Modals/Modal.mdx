---
name: Modal
menu: Components
---

import { Component, Fragment } from 'react';
import { Playground, PropsTable } from 'docz';
import Modal from './Modal';
import Button, { ButtonSet } from '../Buttons';
import Typography from '../Typography';
import { CardHeader, CardBody, CardActions, CardFooter } from '../Card';

# Modal

Modals use cards to display their content, so check out the Card component
for what is allowed in Card content.

You can make a modal not respond to clicks outside the modal or ESC keys
by not handling the `onClose` prop.

## \<Modal /\> component

```js
import Modal from '@clarityhub/unity-web/lib/components/Modal';
```

## Prop Types

### Modal

<PropsTable of={Modal} />

## Examples

### Simple Modal

<Playground>
    {() => {
        class ModalExample extends Component {
            constructor(props) {
                super(props);

                this.state = { open: false };
            }
            render() {
                const { open } = this.state;

                return (
                    <Fragment>
                        <Button onClick={() => this.setState({ open: true })}>
                            Open Modal
                        </Button>

                        <Modal open={open} onClose={() => this.setState({ open: false })}>
                            <CardBody>
                                <Typography type="h3">Card title</Typography>
                                <Typography type="h4">Card subtitle</Typography>
                                <Typography type="text">
                                    Meow Mix
                                </Typography>

                                <CardActions>
                                    <ButtonSet>
                                        <Button
                                            text
                                            onClick={() => this.setState({ open: false })}
                                        >
                                            Close
                                        </Button>
                                        <Button
                                            type="danger"
                                            onClick={() => this.setState({ open: false })}
                                        >
                                            Delete
                                        </Button>
                                    </ButtonSet>
                                </CardActions>
                            </CardBody>
                        </Modal>
                    </Fragment>
                );
            }
        }


        return <ModalExample />;
    }}
</Playground>

### Notification Modal

<Playground>
    {() => {
        class ModalExample extends Component {
            constructor(props) {
                super(props);

                this.state = { open: false };
            }
            render() {
                const { open } = this.state;

                return (
                    <Fragment>
                        <Button onClick={() => this.setState({ open: true })}>
                            Open Modal
                        </Button>

                        <Modal
                            open={open}
                            onClose={() => this.setState({ open: false })}
                            type="notification"
                            center
                        >
                            <CardBody>
                                <Typography noMargin noPadding center color="white" type="h2">
                                    Join our newsletter
                                </Typography>
                                <Typography center color="white" type="text">
                                    Get updates on Clarity Hub, open source,
                                    and more by signing up for our newsletter.
                                </Typography>
                                <CardActions>
                                    <ButtonSet spread>
                                        <Button
                                            text
                                            type="white"
                                            onClick={() => this.setState({ open: false })}
                                        >
                                            No Thanks
                                        </Button>
                                        <Button
                                            type="white"
                                            onClick={() => this.setState({ open: false })}
                                        >
                                            Sign Up
                                        </Button>
                                    </ButtonSet>
                                </CardActions>
                            </CardBody>
                        </Modal>
                    </Fragment>
                );
            }
        }


        return <ModalExample />;
    }}
</Playground>
