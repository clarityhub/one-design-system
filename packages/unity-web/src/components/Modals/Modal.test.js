import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../Button';
import ButtonSet from '../ButtonSet';
import Typography from '../Typography';
import { CardBody, CardActions } from '../Card';
import Modal from './Modal';

describe('<Modal />', () => {
	const instance = {
		getBoundingClientRect() {
			return {
				top: 0,
				left: 0,
				right: 0,
				bottom: 0,
				height: 0,
				outerHeght: 0,
				width: 0,
			};
		},
	};
	function createNodeMock() {
		return instance;
	}

	it('renders a Modal', () => {
		const node = (
			<Modal noPortal open={true} onClose={() => {}}>
				<CardBody>
					<Typography type="h3">Card title</Typography>
					<Typography type="h4">Card subtitle</Typography>
					<Typography type="text">
                        Meow Mix
					</Typography>

					<CardActions>
						<ButtonSet>
							<Button
								text
							>
                                Close
							</Button>
							<Button
								type="danger"
							>
                                Delete
							</Button>
						</ButtonSet>
					</CardActions>
				</CardBody>
			</Modal>
		);
		const tree = renderer.create(node, { createNodeMock }).toJSON();

		expect(tree).toMatchSnapshot();
	});
    
});


