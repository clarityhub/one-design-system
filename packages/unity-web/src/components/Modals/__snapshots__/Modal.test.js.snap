// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> renders a Modal 1`] = `
.emotion-15 {
  -webkit-transition: all .3s ease-out;
  transition: all .3s ease-out;
  -webkit-transition-property: opacity,-webkit-transform;
  -webkit-transition-property: opacity,transform;
  transition-property: opacity,transform;
  -webkit-transform: translate(0,-25%);
  -ms-transform: translate(0,-25%);
  transform: translate(0,-25%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: calc(100% - (.5rem * 2));
  opacity: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 500px;
  margin: 1.75rem auto;
  position: relative;
  width: 100%;
  opacity: 1;
  -webkit-transform: translate(0,0);
  -ms-transform: translate(0,0);
  transform: translate(0,0);
}

@media screen and (prefers-reduced-motion:reduce) {
  .emotion-15 {
    -webkit-transition: none;
    transition: none;
  }
}

.emotion-15 .emotion-14 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  pointer-events: auto;
}

.emotion-13 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-clip: border-box;
  border-radius: 0.25rem;
  box-shadow: 0 4px 6px rgba(50,50,93,.11),0 1px 3px rgba(0,0,0,.08);
  background-color: #FFF;
}

.emotion-13 >:not(.eedi62r0) h3 {
  margin: 0;
  padding: 0;
}

.emotion-13 >:not(.eedi62r0) h4 {
  margin-top: 0.6rem;
}

.emotion-13 >:not(.eedi62r0) p:only-child {
  margin: 0;
}

.emotion-11 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  padding: 1.25rem;
}

.emotion-0 {
  font-family: Futura,'Trebuchet MS',Arial,sans-serif;
  font-weight: 400;
  font-size: 1.3125rem;
  line-height: 1.2;
  padding-bottom: 1rem;
  padding-top: 3rem;
  color: #322439;
}

.emotion-1 {
  font-family: 'Roboto',Arial,sans-serif;
  font-weight: 400;
  font-size: 1.1875rem;
  line-height: 1.2;
  color: #322439;
}

.emotion-2 {
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  color: #322439;
}

.emotion-2 + .emotion-2 {
  margin-top: 1.6rem;
}

.emotion-2 strong,
.emotion-2 b {
  font-weight: 800;
}

.emotion-2 i,
.emotion-2 em {
  font-style: italic;
}

.emotion-9 {
  margin-top: 1rem;
}

.emotion-7 > * + * {
  margin-left: 0.5rem;
}

.emotion-3 {
  overflow: visible;
  display: inline-block;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.35px;
  -moz-letter-spacing: 0.35px;
  -ms-letter-spacing: 0.35px;
  letter-spacing: 0.35px;
  box-shadow: 0 4px 6px rgba(50,50,93,.11),0 1px 3px rgba(0,0,0,.08);
  cursor: pointer;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid transparent;
  padding: .5rem 1.25rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  --text: #566ad2;
  --text-active: #333;
  --bg: transparent;
  --bg-active: transparent;
  --bg-disabled: transparent;
}

.emotion-3:focus,
.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3:focus,
.emotion-3.focused {
  outline: 0;
  box-shadow: 0 0 0 .2rem rgba(0,123,255,.25);
}

.emotion-3:disabled,
.emotion-3.disabled {
  cursor: not-allowed;
  pointer-events: none;
}

.emotion-3:not(:disabled):not(.disabled) {
  cursor: pointer;
}

.emotion-3 svg {
  height: 1rem;
  width: 1rem;
}

.emotion-3.-default,
.emotion-3.-primary,
.emotion-3.-danger,
.emotion-3.-white {
  color: var(--text);
  background-color: var(--bg);
  border-color: var(--bg);
}

.emotion-3.-default.-outline,
.emotion-3.-primary.-outline,
.emotion-3.-danger.-outline,
.emotion-3.-white.-outline {
  border-color: var(--outline,var(--bg));
  color: var(--outline-text,var(--bg));
}

.emotion-3.-default.-text,
.emotion-3.-primary.-text,
.emotion-3.-danger.-text,
.emotion-3.-white.-text {
  color: var(--text-text,var(--bg));
}

.emotion-3.-default:hover,
.emotion-3.-primary:hover,
.emotion-3.-danger:hover,
.emotion-3.-white:hover,
.emotion-3.-default.active,
.emotion-3.-primary.active,
.emotion-3.-danger.active,
.emotion-3.-white.active {
  background-color: var(--bg-active);
  border-color: var(--bg-active);
  color: var(--text-active);
}

.emotion-3.-default:hover.-outline,
.emotion-3.-primary:hover.-outline,
.emotion-3.-danger:hover.-outline,
.emotion-3.-white:hover.-outline,
.emotion-3.-default.active.-outline,
.emotion-3.-primary.active.-outline,
.emotion-3.-danger.active.-outline,
.emotion-3.-white.active.-outline {
  background-color: var(--bg) !important;
  border: 1px solid var(--bg);
  color: var(--outline-text-active,var(--text));
}

.emotion-3.-default:hover.-text,
.emotion-3.-primary:hover.-text,
.emotion-3.-danger:hover.-text,
.emotion-3.-white:hover.-text,
.emotion-3.-default.active.-text,
.emotion-3.-primary.active.-text,
.emotion-3.-danger.active.-text,
.emotion-3.-white.active.-text {
  color: var(--text-text-active,var(--bg-active));
}

.emotion-3.-default:disabled,
.emotion-3.-primary:disabled,
.emotion-3.-danger:disabled,
.emotion-3.-white:disabled,
.emotion-3.-default.disabled,
.emotion-3.-primary.disabled,
.emotion-3.-danger.disabled,
.emotion-3.-white.disabled {
  background-color: var(--bg-disabled);
  border-color: var(--bg-disabled);
  color: var(--text);
}

.emotion-3.-default:disabled.-outline,
.emotion-3.-primary:disabled.-outline,
.emotion-3.-danger:disabled.-outline,
.emotion-3.-white:disabled.-outline,
.emotion-3.-default.disabled.-outline,
.emotion-3.-primary.disabled.-outline,
.emotion-3.-danger.disabled.-outline,
.emotion-3.-white.disabled.-outline {
  border: 1px solid var(--outline-disabled,var(--outline-text-disabled,var(--bg-disabled)));
  color: var(--outline-text-disabled,var(--bg-disabled));
}

.emotion-3.-default:disabled.-text,
.emotion-3.-primary:disabled.-text,
.emotion-3.-danger:disabled.-text,
.emotion-3.-white:disabled.-text,
.emotion-3.-default.disabled.-text,
.emotion-3.-primary.disabled.-text,
.emotion-3.-danger.disabled.-text,
.emotion-3.-white.disabled.-text {
  color: var(--text-text-disabled,var(--bg-disabled));
}

.emotion-3.-primary {
  --text: #FFF;
  --text-active: #FFF;
  --outline-disabled: #C2C2C2;
  --bg: #566ad2;
  --bg-active: rgb(70,90,197);
  --bg-disabled: rgb(114,131,211);
}

.emotion-3.-danger {
  --text: #FFF;
  --text-active: #FFF;
  --outline-disabled: #C2C2C2;
  --bg: #f5365c;
  --bg-active: rgb(233,37,76);
  --bg-disabled: rgb(249,139,161);
}

.emotion-3.-default {
  --text: #333;
  --text-active: #333;
  --outline: #333;
  --outline-text: #333;
  --outline-text-active: #333;
  --outline-text-disabled: #C2C2C2;
  --text-text: #333;
  --text-text-active: #333;
  --text-text-disabled: #C2C2C2;
  --bg: #FFF;
  --bg-active: #FDFDFD;
  --bg-disabled: #F1F1F1;
}

.emotion-3.-white {
  --text: #566ad2;
  --text-active: #333;
  --outline: #FFF;
  --outline-text: #FFF;
  --outline-text-active: #566ad2;
  --outline-text-disabled: #C2C2C2;
  --text-text: #FFF;
  --text-text-active: #FFF;
  --text-text-disabled: #C2C2C2;
  --bg: #FFF;
  --bg-active: #FDFDFD;
  --bg-disabled: #F1F1F1;
}

.emotion-3.-text {
  background-color: transparent !important;
  border: 1px solid transparent;
  box-shadow: none;
}

.emotion-3.-text:hover,
.emotion-3.-text.active {
  border-color: transparent;
}

.emotion-3.-text.disabled,
.emotion-3.-text:disabled {
  border: 1px solid transparent;
}

.emotion-3.-outline {
  background-color: transparent !important;
  border: 1px solid inherit;
  box-shadow: none;
}

.emotion-3.disabled,
.emotion-3:disabled {
  cursor: not-allowed;
}

.emotion-17 {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: rgba(50,36,57,0.6);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 9999;
}

<div>
  <div
    className="emotion-17 emotion-18"
  >
    <div
      className="emotion-15 emotion-16"
    >
      <div
        className="emotion-13 emotion-14"
      >
        <div
          className="emotion-11 emotion-12"
        >
          <h3
            className="emotion-0"
          >
            Card title
          </h3>
          <h4
            className="emotion-1"
          >
            Card subtitle
          </h4>
          <p
            className="emotion-2"
          >
            Meow Mix
          </p>
          <div
            className="emotion-9 emotion-10"
          >
            <div
              className="emotion-7 emotion-8"
            >
              <button
                className="-default -text emotion-3 emotion-4"
                type="button"
              >
                Close
              </button>
              <button
                className="-danger emotion-3 emotion-4"
                type="button"
              >
                Delete
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
